spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural.py

```diff
@@ -0,0 +1,567 @@
+                connections, connection_row_indices, n_rows,
+                post_vertex_slice, n_synapse_types)
+        return self._super.get_static_synaptic_data(
+            connections, connection_row_indices, n_rows, post_vertex_slice,
+            n_synapse_types)
+
+    def get_static_synaptic_data(self, connections, connection_row_indices,
+                                 n_rows, post_vertex_slice,
+                                 n_synapse_types, app_edge, machine_edge):
+        if post_vertex_slice.lo_atom not in self._connections.keys():
+            self._connections[post_vertex_slice.lo_atom] = []
+        self._connections[post_vertex_slice.lo_atom].append(
+            (connections, app_edge, machine_edge))
+        return self._super.get_static_synaptic_data(
+            connections, connection_row_indices, n_rows, post_vertex_slice,
+            n_synapse_types)
+
+    def get_n_words_for_plastic_connections(self, n_connections):
+        try:
+            self._actual_row_max_length = \
+                self._super.get_n_words_for_plastic_connections(n_connections)
+        except Exception:
+            self._actual_row_max_length = \
+                self._super.get_n_words_for_static_connections(n_connections)
+        return self._actual_row_max_length
+
+    def get_n_words_for_static_connections(self, n_connections):
+        self._actual_row_max_length = \
+            self._super.get_n_words_for_static_connections(n_connections)
+        return self._actual_row_max_length
+
+    def get_n_synapses_in_rows(self, pp_size, fp_size=None):
+        if fp_size is not None:
+            return self._super.get_n_synapses_in_rows(pp_size, fp_size)
+        return self._super.get_n_synapses_in_rows(pp_size)
+
+    def read_plastic_synaptic_data(self, post_vertex_slice, n_synapse_types,
+                                   pp_size, pp_data, fp_size, fp_data):
+        return self._super.read_plastic_synaptic_data(
+            post_vertex_slice, n_synapse_types, pp_size, pp_data, fp_size,
+            fp_data)
+
+    def read_static_synaptic_data(self, post_vertex_slice, n_synapse_types,
+                                  ff_size, ff_data):
+        return self._super.read_static_synaptic_data(
+            post_vertex_slice, n_synapse_types, ff_size, ff_data)
+
+    def get_n_fixed_plastic_words_per_row(self, fp_size):
+        return self._super.get_n_fixed_plastic_words_per_row(fp_size)
+
+    def get_n_plastic_plastic_words_per_row(self, pp_size):
+        return self._super.get_n_plastic_plastic_words_per_row(pp_size)
+
+    def are_weights_signed(self):
+        return self._super.are_weights_signed()
+
+    def get_vertex_executable_suffix(self):
+        name = self._super.get_vertex_executable_suffix()
+        name += "_structural"
+        return name
+
+    def get_n_static_words_per_row(self, ff_size):
+        return self._super.get_n_static_words_per_row(ff_size)
+
+    def is_same_as(self, synapse_dynamics):
```

Comment:
using private variables! make properties and thus use public variables.

Reply:
also, why not using __equals__

Reply:
The conventional is-same-as operator in Python is written `==`…